sdk/cosmos/azure-cosmos/azure/cosmos/_location_cache.py

```diff
@@ -159,23 +159,22 @@ def should_refresh_endpoints(self):  # pylint: disable=too-many-return-statement
 
             should_refresh = self.use_multiple_write_locations and not self.enable_multiple_writable_locations
 
-            if most_preferred_location:
-                if self.available_read_endpoint_by_locations:
-                    most_preferred_read_endpoint = self.available_read_endpoint_by_locations[most_preferred_location]
-                    if most_preferred_read_endpoint and most_preferred_read_endpoint != self.read_endpoints[0]:
-                        # For reads, we can always refresh in background as we can alternate to
-                        # other available read endpoints
-                        return True
-                else:
+            if most_preferred_location and most_preferred_location in self.available_read_endpoint_by_locations:
```

Comment:
To double check - is this purely a refactoring change?

Reply:
it's a bugfix truthfully - since before we were not checking if the most_preferred_region was in the available endpoints, we were leaving ourselves open to trying to access a missing value directly (which would throw a KeyError due to it missing in the dictionary).

Reply:
Same with L177.